### PR TITLE
Update release template

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
- Auto label as `epic`
- Add an HTML comment to prevent future people (me) copy/pasting instead of using the template.
- Add important dates section
- Clean up check lists based on the recent release tickets:
  - https://github.com/stackabletech/issues/issues/488
  - https://github.com/stackabletech/issues/issues/557

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -82,14 +93,16 @@ Marketing tasks can now reference published documentation.
...
 - [ ] Send announcement to Kubernetes Podcast (Ping Lars)
 - [ ] Send announcement to Heiser
+- [ ] Ping the stackable-ionos-tech channel or anyone responsible once all tags are created
 ```
 
 ```[tasklist]
 ### Post-release tasks
 - [ ] Bump Rust version. This can be done [in this file](https://github.com/stackabletech/operator-templating/blob/main/config/rust.yaml) by changing `rust_version` and also for the ubi base image [here](https://github.com/stackabletech/docker-images/blob/main/ubi8-rust-builder/Dockerfile#L25). Please be aware that this action will change it for all repositories at the same time (when merging the templating PRs).
-- [ ] Run renovate manually by starting this action: https://github.com/stackabletech/ci/blob/main/.github/workflows/renovate.yml
+- [ ] ~~Run renovate manually by starting this action: https://github.com/stackabletech/ci/blob/main/.github/workflows/renovate.yml~~ (Doesn't work. Renovate is not automatable)
```
Comment (razvan) [inline .github/ISSUE_TEMPLATE/release.md:102]: maybe we can remove it completely

**Review by razvan — APPROVED** [review]
lgtm

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -41,13 +51,14 @@ This will not be so crucial with release branches, but is nonetheless sensible a
 - [ ] Create release tag(s) for operators (see stackable-utils for scripts to create tags)
 - [ ] Create release tag for stackable-cockpit (optional, highly experimental, requires manual tag creation)
 - [ ] Update changelogs in main branches (see stackable-utils for script to do this)
-- [ ] Generate CRD docs [website](https://crds.stackable.tech/) for the new release by following these [instructions](https://github.com/stackabletech/crddocs)
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/release.md:44]: Done in f54b7cf

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/release.md]: Done in f54b7cf

**Pull request merged**